dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/importer/context/CategoryOptionComboSupplier.java

```diff
@@ -118,4 +105,43 @@ else if ( programCatCombo != null && isNotEmpty( attributeCatOptions ) )
         return eventToCocMap;
     }
 
+    private CategoryOptionCombo getCategoryOptionCombo( Program program, Event event, ProgramStageInstance psi,
+        IdScheme idScheme )
+    {
+        final CategoryCombo programCatCombo = program.getCategoryCombo();
+        String aoc = event.getAttributeOptionCombo();
+        String attributeCatOptions = event.getAttributeCategoryOptions();
+
+        /*
+         * Event create request contain aoc information in payload but for event
+         * update aoc should be fetched from ProgramStageInstance
+         */
+        if ( psi != null && aoc == null )
```

Comment:
It would be good to have this case covered in the tests

Reply:
test added